maven_plugin/src/main/java/io/github/chains_project/maven_lockfile/data/RepositoryId.java

```diff
@@ -0,0 +1,55 @@
+package io.github.chains_project.maven_lockfile.data;
+
+import java.util.Objects;
+
+/**
+ * This class represents a repository id for an artifact.
+ * The repository id contains the id for the repository object in the pom.
+ * It is a string that is not null or empty. It is immutable.
+ */
+public class RepositoryId implements Comparable<RepositoryId> {
+    public static RepositoryId of(String repositoryId) {
+        // RepositoryId must be non-null.
+        String checked = Objects.requireNonNull(repositoryId, "resolvedUrl cannot be null");
+        return new RepositoryId(checked);
+    }
+
+    public static RepositoryId Default() {
+        return new RepositoryId("central");
+    }
+
```

Comment:
`None()` option for providing an empty string representation.

```suggestion
  public static RepositoryId None() {
      return new RepostioryId("");
  }
```